### PR TITLE
Improve phone input UI

### PR DESCRIPTION
**Pull request by siv-io**
[FIX] #8251 

Changes made as suggested by @Bonapara. 

image
image
image
image

For the `The country code should be Tertiary instead of Primary ` task,
the library "react-phone-number-input" doesn't provide any out of the box functionality to style the country code. 

If the feature **needs** to be implemented here are the possible solution/workarounds:

1. Finding a more customizable library that allows to change the style of the country code

OR 

2. Implement custom country selection (😰...)

OR

3.  The lib allows a custom input element and a provides a function (onCountryChange) that triggers whenever the country changes (a country can be changed in two ways- 1. When the user deliberately chooses it from dropdown OR 2. Changes the code in the input)

We'll have to get the length of the country code and then style the first X digits in the custom input field...

image
image

Let me know if someone has a better approach.

### CODE REVIEW

**Comment by Bonapara** [comment]
Thanks @Hitarthsheth07 for your contribution!

# Country Code

Let's keep it to color:primary then

# Padding

### Current

CleanShot 2024-11-01 at 12 51 05@2x

The right padding is too large.

### Desired

It should be 4px so that the icon button's top, left, and bottom padding are equal.

CleanShot 2024-11-01 at 12 52 40@2x

# Dropdown menu

### Current

CleanShot 2024-11-01 at 13 13 54@2x

Width is 50

### Desired

Width should be 44

CleanShot 2024-11-01 at 13 15 12@2x

**Comment by Hitarthsheth07** [comment]
Thanks for the feedback @Bonapara 

Here are the changes:

## Dropdown menu

I've added the left padding 8px and right padding 4x. 
Icons and gap now have proper attributes. 

The overall width is still 44.8px because it also takes into account the border.
image

While in figma we are using _gap_ to show border, so the overall width of the left container remains same. 

https://www.loom.com/share/11ddb9ba2402479bbdb6b306e30215cc?sid=5b0da58a-42f3-4bf8-8fa8-21bf870c72e2

This could be solved by changing the bg color of the parent and children div and then adding gap to them, similar approach to what you're doing in figma. 
**Let me know if want to implement this.**

## Right Add icon

As suggested padding on all the sides is even now. 
image

**Comment by Bonapara** [comment]
Hi @Hitarthsheth07, regarding the padding, I'm referring to the gap between the icon and input edge:

3f9df329bb697621da0eb49e6238bf34cbb4d7f06faaec30c9e595585dc0c59f

Also your PR seems to cause the picker to freeze when I click on the three dots menu:

https://github.com/user-attachments/assets/dc941c6a-d751-4e66-8b34-bc959d69954d

**Comment by Hitarthsheth07** [comment]
> Hi @Hitarthsheth07, regarding the padding, I'm referring to the gap between the icon and input edge:
> 
> 
> 
> 3f9df329bb697621da0eb49e6238bf34cbb4d7f06faaec30c9e595585dc0c59f
> 
> 
> 
> Also your PR seems to cause the picker to freeze when I click on the three dots menu:
> 
> 
> 
> https://github.com/user-attachments/assets/dc941c6a-d751-4e66-8b34-bc959d69954d
> 
> 
> 
> 

Ahh got it!

**Comment by Bonapara** [comment]
@Hitarthsheth07 The bug was actually caused by something else and has been fixed on the main branch. I updated your branch to the main, and the bug is gone. Let's forget about this issue and focus on fixing the icon padding!

**Comment by Hitarthsheth07** [comment]
> @Hitarthsheth07 The bug was actually caused by something else and has been fixed on the main branch. I updated your branch to the main, and the bug is gone. Let's forget about this issue and focus on fixing the icon padding!

On it, will push the commit in an hour or two.

**Comment by Hitarthsheth07** [comment]
image

@Bonapara

File: packages/twenty-front/src/modules/ui/input/components/internal/phone/components/PhoneCountryPickerDropdownButton.tsx [hunk 1]
```diff
@@ -95,7 +103,9 @@ export const PhoneCountryPickerDropdownButton = ({
         <StyledDropdownButtonContainer isUnfolded={isDropdownOpen}>
           <StyledIconContainer>
             {selectedCountry ? <selectedCountry.Flag /> : <IconWorld />}
-            <IconChevronDown size={theme.icon.size.sm} />
+            <StyledCheveronIconContainer>
```
Comment (Bonapara) [inline packages/twenty-front/src/modules/ui/input/components/internal/phone/components/PhoneCountryPickerDropdownButton.tsx:106]: Why did you add a styled container?

**Comment by Bonapara** [comment]
image

It seems I have a 1px gap with the cell. Is this the case for you as well? It should be on top of the cell border.

File: packages/twenty-front/src/modules/ui/input/components/internal/phone/components/PhoneCountryPickerDropdownButton.tsx (hunk 1 above)
Comment (Hitarthsheth07) [reply packages/twenty-front/src/modules/ui/input/components/internal/phone/components/PhoneCountryPickerDropdownButton.tsx:106]: The chevron down icon is not sized as per the figma file, the only way to change the size is by adding a container

**Comment by Hitarthsheth07** [comment]
image

@Bonapara the gap between cell and dialog box is already there in this version. 
Should I push a commit to change it?

**Comment by Bonapara** [comment]
You're right! I think we will fix it in the release quick fixes, so no need thanks!

**Comment by Hitarthsheth07** [comment]
> You're right! I think we will fix it in the release quick fixes, so no need thanks!

Sounds good! Can this be merged and closed 

Thanks.

**Comment by Bonapara** [comment]
@lucasbordeau if you can do the review - lgtm from a product perspective

**Comment by Hitarthsheth07** [comment]
@lucasbordeau any updates on this PR?

**Review by lucasbordeau — APPROVED** [review]
